syntax = "proto3";

package prediction;

service Prediction {
  rpc PredictMultiple (PredictRequest) returns (PredictResponse);
}

message PredictRequest {
  repeated TaskInfo tasks = 1;
}

message PredictResponse {
  repeated float prediction_results = 1;
}

message TaskInfo {
  string due_date = 1;
  float estimated_hours = 2;
  string completed_on = 3;
  float current_labor_hours = 4;
  string category_name = 5;
  string site_name = 6;
  string created_at = 7;
  bool is_completed_on_time = 8;
}
